### DISCUSSION TITLE
UE ping to the core network (open5gs)

### DISCUSSION THREAD

**Discussion by initChat** [srsran/srsRAN_Project#690] [General Help] [answered]
My system the UE already connect to the network. **However, I cannot ping between the UE (Raspberry Pi with Quectel board) and the core network (open5gs).**

I have `ogstun` as interface on open5gs (core network) with ip address `10.45.0.1`
```
ogstun: flags=4241<UP,POINTOPOINT,NOARP,MULTICAST>  mtu 1400
        inet 10.45.0.1  netmask 255.255.0.0  destination 10.45.0.1
        unspec 00-00-00-00-00-00-00-00-00-00-00-00-00-00-00-00  txqueuelen 500  (UNSPEC)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
```
and I connect the UE to the network and receive ip as `10.45.0.2`
```
>> sudo mmcli -m 1 --simple-connect="apn=internet"
>> sudo udhcpc -q -f -i wwan0
udhcpc: started v1.35.0
udhcpc: broadcasting discover
udhcpc: broadcasting select for 10.45.0.2 server 10.45.0.1
udhcpc: lease of 10.45.0.2 obtained from 10.45.0.1, lease time 7200
```
However, the core network and the UE cannot ping to each other. I set Wireshark on the core network (10.45.0.1).
When I ping from core network -> UE. The result on wireshark has request and reply, but nothing on pinf terminal.
![Screenshot from 2024-06-26 15-17-00](https://github.com/srsran/srsRAN_Project/assets/40858099/4282ea03-f012-477c-b2c3-dc928a0b099c)
```
>> ping 10.45.0.2
PING 10.45.0.2 (10.45.0.2) 56(84) bytes of data.
^C
--- 10.45.0.2 ping statistics ---
8 packets transmitted, 0 received, 100% packet loss, time 7172ms
```
And when I ping from UE -> core network. There is no result at all on both ping terminal and wireshark (no response found!).
![Screenshot from 2024-06-26 15-17-11](https://github.com/srsran/srsRAN_Project/assets/40858099/efe929dc-5d24-40a1-9aca-f00ef7479087)
```
>> ping 10.45.0.1
PING 10.45.0.1 (10.45.0.1) from 10.45.0.2 wwan0: 56 (84) bytes of data.
^C
6 packets transmitted, 0 received, 100% packet loss, time 5107ms
```

**What did I do wrong? Why I cannot ping between the UE and the core network?**

**Log**
gnb_log - [gnb_log.zip](https://github.com/user-attachments/files/15984810/gnb_log.zip)
wireshark on the core network side- [open5gs_wireshark.zip](https://github.com/user-attachments/files/15984819/open5gs_wireshark.zip)
open5gs_log -[open5gs_log.zip](https://github.com/user-attachments/files/15984830/open5gs_log.zip)

**System**
core network - open5GS run on 5gc with configuration file sample.yaml
RAN - srsRAN_Project
SDR - USRP B210
UE - RPI3 with Quectel RM510Q-GL
Computer - Thinkpad L14, Ubuntu 22.04

**Comment by pgawlowicz** [comment]
does the core network and gnb run on the same PC? 

Could you print routing tables at the core network PC and at UE? (`route -n`)

**Reply by initChat** [reply]
yes the core network and gnb run on the same PC.
on the core network PC
```
>> route -n
Kernel IP routing table
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
0.0.0.0         192.168.8.1     0.0.0.0         UG    100    0        0 enp0s31f6
0.0.0.0         10.221.40.1     0.0.0.0         UG    600    0        0 wlp9s0
10.45.0.0       0.0.0.0         255.255.0.0     U     0      0        0 ogstun
10.53.1.0       0.0.0.0         255.255.255.0   U     0      0        0 br-98a60bc60f67
10.221.40.0     0.0.0.0         255.255.248.0   U     600    0        0 wlp9s0
169.254.0.0     0.0.0.0         255.255.0.0     U     1000   0        0 ogstun
172.17.0.0      0.0.0.0         255.255.0.0     U     0      0        0 docker0
172.19.1.0      0.0.0.0         255.255.255.0   U     0      0        0 br-4d4aa4c50059
192.168.8.0     0.0.0.0         255.255.255.0   U     100    0        0 enp0s31f6
```
on the UE
```
>> route -n
Kernel IP routing table
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
0.0.0.0         10.45.0.1         0.0.0.0         UG    0    0        0 wwan0
10.45.0.0       0.0.0.0         255.255.0.0     U     0      0        0 wwan0
```

**Reply by pgawlowicz** [reply]
strange. Maybe enabling ip forwarding ?

**Reply by initChat** [reply]
already did on core network
```
### Enable IPv4/IPv6 Forwarding
sudo sysctl -w net.ipv4.ip_forward=1
sudo sysctl -w net.ipv6.conf.all.forwarding=1

### Add NAT Rule
sudo iptables -t nat -A POSTROUTING -s 10.45.0.0/16 ! -o ogstun -j MASQUERADE
sudo ip6tables -t nat -A POSTROUTING -s 2001:db8:cafe::/48 ! -o ogstun -j MASQUERADE

sudo ufw disable

### Prevent UE's from connecting to the host on which UPF is running
sudo iptables -I INPUT -s 10.45.0.0/16 -j DROP
sudo ip6tables -I INPUT -s 2001:db8:cafe::/48 -j DROP
```

**Reply by pgawlowicz** [reply, marked as answer]
what about those rules:
```
### Prevent UE's from connecting to the host on which UPF is running
sudo iptables -I INPUT -s 10.45.0.0/16 -j DROP
sudo ip6tables -I INPUT -s 2001:db8:cafe::/48 -j DROP
```

Isn't is cutting out the ping?

**Reply by pgawlowicz** [reply]
@pchat-imm any update on this issue?

**Reply by initChat** [reply]
yes, I do it according to your guide and i can ping my core network now. Thank you

**Comment by dhiaboujebha** [comment]
@pgawlowicz i have the same issue in my setup right now. We are using RM500-Q. It's getting attached and has an ip address over the interface wwan0. But, i cannot ping. I tried the solutions written above but none of them solved that. 
Any idea?
